src/transaction/utils.ts

```diff
@@ -63,25 +64,35 @@ function prepareTransaction(txJSON: any, api: RippleAPI,
     const multiplier = instructions.signersCount === undefined ? 1 :
       instructions.signersCount + 1
     if (instructions.fee !== undefined) {
+      const fee = new BigNumber(instructions.fee)
+      if (fee.greaterThan('2') && instructions.allowHighFee !== true) {
+        const errorMessage = `Fee of ${fee.toString(10)} XRP exceeds ` +
+          'soft limit of 2 XRP. To use this fee, set ' +
+          '`instructions.allowHighFee = true`.'
+        throw new ValidationError(errorMessage)
+      }
       txJSON.Fee = scaleValue(common.xrpToDrops(instructions.fee), multiplier)
       return Promise.resolve(txJSON)
     }
     const cushion = api._feeCushion
```

Comment:
This isn't new, but we set `cushion` to `api._feeCushion` both here and inside `getFee`.

Reply:
I think this is ok.